### PR TITLE
Added Sharding section

### PR DESCRIPTION
**Pull request by rcbensley**
Adding some common sharding options.

### CODE REVIEW

**Review by federico-razzoli — CHANGES_REQUESTED** [review]
Some notes:
- Modify the intro text to make it clear that there is no complete sharding solution for MariaDB, but we can shard the data with a dedicated storage engine and route the queries to the right shards with a proxy.
- Keep the items in alphabetical order.
- As "Project" I'd indicate the software, eg, ProxySQL. I'd add a Notes column to briefly explain how it's useful for sharding (eg, routes queries based on a regexp).

File: list-dba.md [hunk 1]
```diff
@@ -103,13 +103,14 @@ In the list below, SQL-aware means that a proxy understands SQL statements, and
...
-|---------------------------------------------------------------------|------------------------------------|------------------------|------------------------------------------------------------|---------|
-| [Sharding in ProxySQL](https://proxysql.com/documentation/how-to-setup-proxysql-sharding/) | YES         | External               | GPLv3, Part of ProxySQL                                    |         |
-| [MaxScale SchemaRouter](https://mariadb.com/kb/en/mariadb-maxscale-24-schemarouter/) | YES               | External               | Commercial, part of Maxscale                               |         |
-| [Spider](https://mariadb.com/kb/en/spider/)                         | YES                                | Internal               | GPL, a MariaDB Storage Engine                              |         |
+Sharding solutions require evalation depending in your applications requirements. Some solutions are internal and shard with the application being unaware, or externally use rules and defititions which your application may need to understand.
```
Comment (federico-razzoli) [inline list-dba.md]: Spider and proxies might be parts of the same sharded setup, but then you'll have to configure them, write scripts to add/remove/resize shards, possibly use replication to distribute some shared data to all the shards, etc. So I'd just write that a complete sharding solution doesn't exist, there are components that solve some aspects of the problem.

**Review by federico-razzoli — CHANGES_REQUESTED** [review]

File: list-dba.md [hunk 2]
```diff
@@ -100,6 +101,18 @@ In the list below, SQL-aware means that a proxy understands SQL statements, and
 - [HAProxy](https://www.haproxy.org/) - Open source, TCP proxy. Not SQL-aware.
   - [haproxy-galera](https://github.com/matthanley/haproxy-galera) - MariaDB Galera health check script for HAProxy.
 
+## Sharding
+
+Sharding solutions require evalation depending on your applications requirements. Some solutions are internal and shard with the application being unaware, or externally using rules and defititions which your application may need to understand or even create and manage.
+
+| Project                                                             | MariaDB Support | License / Platform | Notes   |
+|---------------------------------------------------------------------|-----------------|--------------------|---------|
+| [CONNECT](https://mariadb.com/kb/en/using-connect-partitioning-and-sharding/) | YES   | GPLv2              | A storage engine. An outward table can be partitioned over serveral remote tables represented as partitions. |
```
Comment (federico-razzoli) [inline list-dba.md:110]: It's true that CONNECT has this feature, but I wouldn't consider it for this purpose. Better to use SPIDER.

**Review by federico-razzoli — CHANGES_REQUESTED** [review]

**Comment by rcbensley** [comment]
Added vitess sharding docs which is more complete than other solutions but not magic by itself.

File: list-dba.md (hunk 2 above)
Comment (rcbensley) [reply list-dba.md]: Removed the engine from sharding

File: list-dba.md [hunk 3]
```diff
@@ -100,6 +101,18 @@ In the list below, SQL-aware means that a proxy understands SQL statements, and
...
+
+| Project                                                             | MariaDB Support | License / Platform | Notes   |
+|---------------------------------------------------------------------|-----------------|--------------------|---------|
+| [MaxScale](https://mariadb.com/kb/en/mariadb-maxscale-24-schemarouter/) | YES         | Commercial         | A query and connection router that is part of Maxscale. |
+| [ProxySQL](https://proxysql.com/documentation/how-to-setup-proxysql-sharding/) | YES  | GPLv3              | Sharding in ProxySQL by User, Schema or Data. Sharding is based on rules which pattern match on incoming queries. |
+| [Spider](https://mariadb.com/kb/en/spider/)                         | YES             | GPLv2              | A storage engine for table definition shards and connections to split tables among several servers. |
+| [Vitess](https://vitess.io/docs/19.0/reference/features/sharding/)  | Yes             | Apache 2.0         | Vitess is external to the database engine and also has migration strategies built in. |
```
Comment (federico-razzoli) [inline list-dba.md:113]: But it doesn't support MariaDB. It did for some time, but that support has EOLed.
https://vitess.io/docs/archive/13.0/overview/supported-databases/

**Review by federico-razzoli — CHANGES_REQUESTED** [review]

File: list-dba.md (hunk 3 above)
Comment (rcbensley) [reply list-dba.md]: Oh that's a shame. Removed.

**Review by federico-razzoli — APPROVED** [review]

**Pull request merged**